backend/src/api/routers/chains.py

```diff
@@ -21,11 +23,33 @@ class UserInput(BaseModel):
 
 
 load_dotenv()
-use_cuda = os.getenv("USE_CUDA")
 
-llm = ChatGoogleGenerativeAI(model="gemini-pro", temperature=1)
-hf_embdeddings = "BAAI/bge-large-en-v1.5"
-hf_reranker = "BAAI/bge-reranker-base"
+required_env_vars = [
+    "USE_CUDA",
+    "GEMINI_TEMP",
+    "HF_EMBEDDINGS",
+    "HF_RERANKER",
+    "GOOGLE_GEMINI",
+]
+
+if any(os.getenv(var) is None for var in required_env_vars):
+    raise ValueError("One or more environment variables are not set.")
+
+use_cuda: bool = os.getenv("USE_CUDA")
```

Comment:
This is not robust. Even if you `USE_CUDA` is "false" it will still read as True.

Reply:
You need to enforce default values as well. Look into `os.getenv`, they allow for defaults.